apis/src/main/kotlin/org/yapp/apis/auth/strategy/AppleAuthStrategy.kt

```
@@ -0,0 +1,83 @@
+            val decodedPayload = Base64.getUrlDecoder().decode(payload)
+            val payloadJson = String(decodedPayload)
+
+            // In a real implementation, you would use a JSON library to parse the payload
+            // For simplicity, we'll just extract the required fields manually
+            val sub = extractField(payloadJson, "\"sub\":\"", "\"")
+                ?: throw AuthException(AuthErrorCode.SUBJECT_NOT_FOUND)
+            val email = extractField(payloadJson, "\"email\":\"", "\"")
+            val name = extractField(payloadJson, "\"name\":\"", "\"")
+
+            return AppleIdTokenPayload(sub, email, name)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)